pkg/clockcache/cache_bench_test.go

```diff
@@ -236,6 +237,74 @@ func BenchmarkInsertConcurrent(b *testing.B) {
 	}
 }
 
+// BenchmarkCacheFalseSharing is a benchmark to measure the effect of the false
+// sharing of cache entries. In the clockcache.element struct, we introduce
+// padding to avoid false sharing.
+// The principle behind this benchmakr is simple: construct a cache with two
+// entries, and start two goroutines which each clobber one of the cache values
+// over and over again.
+// If there is false sharing, it should be measurable by toggling the padding
+// on and off, and measuring the difference in output of this benchmark.
+//
+// At the time of writing, this code was tested on an M1 MacBook Pro, where the
+// advantage obtained by introducing padding is approximately 16%:
+//     go test -bench=BenchmarkCacheFalseSharing -cpu=2 -count=10 > no-padding.txt
+//     go test -bench=BenchmarkCacheFalseSharing -cpu=2 -count=10 > padding.txt
+//     benchstat no-padding.txt padding.txt
+//       name                 old time/op    new time/op    delta
+//       CacheFalseSharing-2     230ns ± 6%     193ns ±19%  -16.09%  (p=0.001 n=10+9)
+//
+// Note: This benchmark _must_ be run with the `-cpu=2` argument, to ensure
+// that each goroutine ends up on a different CPU.
+func BenchmarkCacheFalseSharing(b *testing.B) {
+	keys := make([]int, 2)
+	vals := make([]int, 2)
+
+	for i := 0; i < 2; i++ {
+		keys[i], vals[i] = i, i
+	}
```

Comment:
Are these lines are added to avoid compiler optimising this routine?

Reply:
No, actually this was kinda just "noise" left over from copy-pasting another benchmark. I've taken another pass at this benchmark and simplified it a little.